### PR TITLE
DAOS-8224 test: Update container/snapshot_aggregation.py

### PR DESCRIPTION
**Pull request by mjmac**
The test was still relying on the old-style scm/nvme pool
query results. Modifies the pool query response to include
the media type for each storage tier so that consumers
can reliably categorize the results.

Also fixes some other bugs in the new daos tool:
  * Snapshot destroy was being called with a null container handle (DAOS-8001)
  * Snapshot destroy didn't support the --snap argument

### CODE REVIEW

**Comment by mjmac** [comment]
One failure: [DAOS-8233](https://jira.hpdd.intel.com/browse/DAOS-8233)

Verified that the PR fixes the test: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6356/9/testReport/FTEST_container/SnapshotAggregation/

File: src/tests/ftest/util/daos_utils_base.py [hunk 1]
```diff
@@ -318,6 +318,7 @@ def __init__(self):
                 super().__init__("destroy-snap")
                 self.epc = FormattedParameter("--epc={}")
                 self.epcrange = FormattedParameter("--epcrange={}")
+                self.snap = FormattedParameter("--snap={}")
```
Comment (shimizukko) [inline src/tests/ftest/util/daos_utils_base.py:321]: I assume --snap will be added.
```
[mkano@wolf-32 pool]$ daos container destroy-snap --help
Usage:
  daos [OPTIONS] container destroy-snap [destroy-snap-OPTIONS] [<pool name or UUID>] [<container name or UUID>]

Application Options:
      --debug                         enable debug output
      --verbose                       enable verbose output (when applicable)
  -j, --json                          enable JSON output

Help Options:
  -h, --help                          Show this help message

[destroy-snap command options]
      -G, --sys-name=                 DAOS system name
      -p, --pool=                     pool UUID (deprecated; use positional arg)
      -d, --path=                     unified namespace path
      -c, --cont=                     container UUID (deprecated; use positional arg)
      -e, --epc=                      snapshot epoch to delete
      -r, --epcrange=                 range of snapshot epochs to delete
```

**Review by shimizukko — DISMISSED** [review]

**Review by saurabhtandan — DISMISSED** [review]
Test code LGTM

File: src/tests/ftest/util/daos_utils_base.py (hunk 1 above)
Comment (mjmac) [reply src/tests/ftest/util/daos_utils_base.py:321]: Yes, it's added as part of this PR.

File: src/control/cmd/daos/snapshot.go [hunk 2]
```diff
@@ -66,19 +67,45 @@ func (cmd *containerSnapshotDestroyCmd) Execute(args []string) error {
...
+			return errors.New("cannot specify both snapshot epoch range and epoch")
+		}
+
 		ap.epcrange_begin = cmd.EpochRange.Begin
 		ap.epcrange_end = cmd.EpochRange.End
+	default:
+		return errors.New("must specify one of snapshot name or epoch or epoch range")
+	}
+
+	cleanup, err := cmd.resolveAndConnect(ap)
```
Comment (kjacque) [inline src/control/cmd/daos/snapshot.go:104]: Not sure which will land first, but this will conflict with #6412.

**Review by kjacque — DISMISSED** [review]

**Review by saurabhtandan — APPROVED** [review]
LGTM

**Review by kjacque — APPROVED** [review]

**Pull request merged**